extensions/wikia/LookupContribs/SpecialLookupContribs_hooks.php

```diff
@@ -1,30 +1,28 @@
-<?PHP
-
-/*
-	this practically does one thing: unsets memcached keys on article save
-*/
-
-/* grapple them hooks */
-global $wgHooks;
-
-/* Run after user really saved an article, and only if we _use_ memcache */
-if ( !LOOKUPCONTRIBS_NO_CACHE ) {
-	$wgHooks['ArticleSaveComplete'][] = 'LookupContribsHooks::ArticleSaveComplete';
-}
-$wgHooks['ContributionsToolLinks'][] = 'LookupContribsHooks::ContributionsToolLinks';
+<?php
 
+/**
+ * Class LookupContribsHooks
+ */
 class LookupContribsHooks {
+
+	/**
+	 * Clear all caches created in LookupContribsCore.  Note that there is one other cache used in that
+	 * class but it includes limit and offset values as part of the key and can't be cleared without implementing
+	 * a shared key system.  For now that cache has a TTL so the data will clear eventually.
+	 *
+	 * @param $article
+	 * @param User $user
+	 *
+	 * @return bool
+	 */
 	static public function ArticleSaveComplete ( $article, User $user ) {
-		global $wgDBname, $wgMemc, $wgSharedDB, $wgUser ;
-		/* unset the key for this user on this database */
-		$username = $user->getName () ;
-		$wgMemc->delete ( "$wgSharedDB:LookupContribs:normal:$username:$wgDBname" ) ;
-		$wgMemc->delete ( "$wgSharedDB:LookupContribs:final:$username:$wgDBname" ) ;
```

Comment:
I looked as far back as Aug 26, 2008 (https://github.com/Wikia/app/commit/cf58c10fdc79fc97253e80fc970dbea2b670d243) and I could not find anywhere that actually set these memcached keys.  I can tell that `normal` and `final` are values used in `LookupContribsCore::mMode` and there is a memkey that uses `mMode`, but it looks nothing like this key.  This is definitely a great example of why one should never build their own cache keys rather than using a common method.

I updated this to clear the only `LookupContribsCore` cache possible.  The other cache value includes limit and offset values and would require clearing an impossible to know number of keys.  In this case I think its fine to let the TTL expire this key, which is what has been happening for at least 7 years anyway.